File: Data Structure I/python/Day 2/merge_sorted_array.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        temp=[0]*len(nums1)
        i=j=k=0
        while(i<m and j<n):
            if(nums1[i]<nums2[j]):
                temp[k]=nums1[i]
                k+=1
                i+=1
            else:
                temp[k]=nums2[j]
                k+=1
                j+=1
        while(i<m):
             temp[k]=nums1[i]
             k+=1
             i+=1
        while(j<n):
            temp[k]=nums2[j]
            k+=1
            j+=1
        k=0
        while(k<len(nums1)):
            nums1[k]=temp[k]
            k+=1
        

File: Data Structure I/java/Day 2/MergeSortedArray.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int[] temp = new int[nums1.length];
        int i=0,j=0,k=0;
        while(i<m&&j<n){
            if(nums1[i]<nums2[j]){
                temp[k++]=nums1[i++];
            }else{
                temp[k++]=nums2[j++];
            }
        }
        while(i<m){
             temp[k++]=nums1[i++];
        }
        while(j<n){
            temp[k++]=nums2[j++];
        }
        k=0;
        while(k<nums1.length){
            nums1[k]=temp[k];
            k++;
        }
    }
}